src/shared_utilities/drivers.jl

```diff
@@ -516,3 +516,51 @@ function vapor_pressure_deficit(T_air, P_air, q_air, thermo_params)
     )
     return es - ea
 end
+
+"""
+    initialize_drivers(a::PrescribedAtmosphere{FT}, r::AbstractRadiativeDrivers{FT}, coords) where {FT}
+
+Creates and returns a NamedTuple with the cache variables required by a `PrescribedAtmosphere` and a`PrescribedRadiativeFluxes`.
+"""
+function initialize_drivers(
+    a::PrescribedAtmosphere{FT},
+    r::PrescribedRadiativeFluxes{FT},
+    coords,
+) where {FT}
+    keys = (:P_liq, :P_snow, :T, :P, :u, :q, :c_co2, :SW_d, :LW_d, :θs)
+    types = ([FT for k in keys]...,)
+    domain_names = ([:surface for k in keys]...,)
+    model_name = :drivers
+    vars =
+        ClimaLSM.initialize_vars(keys, types, domain_names, coords, model_name)
+    return vars
```

Comment:
This code is shared between both methods of this function. If we expect we'll add more methods, maybe we could move this to a different function (e.g. `get_driver_vars(keys, coords)`) that they all call, but if we just have two methods I think duplicating is okay

Reply:
I kept this part the same, but only because it's possible we might have different behavior in the future - e.g., if we ever want to run with more wavelength bands, we might want the type of `SW_d` to be a Tuple, with a value for each wavelength portion (instead of all FT). 

If on the other hand more methods are added in the future that call the same code, we can extract this into a function like you suggested then.

Reply:
BTW: I tried to make it a bit more modular by instead defining `initialize_drivers(atmos, coords)` and `initialize_drivers(radiation, coords)`. Then `initalize_drivers(atmos, radiation, coords)` composes them by calling each on its own.